tests/test_ic_od_to_kvp_wrapper/test_vqa_as_kvp.py

```diff
@@ -0,0 +1,34 @@
+import unittest
+
+from tests.test_fixtures import VQATestFixtures
+from vision_datasets.common.constants import DatasetTypes
+from vision_datasets.key_value_pair.manifest import KeyValuePairLabelManifest
+from vision_datasets.visual_question_answering import VQAAsKeyValuePairDataset
+
+
+class TestVQAAsKeyValuePairDataset(unittest.TestCase):
+    def test_vqa_to_kvp(self):
+        sample_vqa_dataset, tempdir = VQATestFixtures.create_a_vqa_dataset()
+        with tempdir:
+            kvp_dataset = VQAAsKeyValuePairDataset(sample_vqa_dataset)
+
+            self.assertIsInstance(kvp_dataset, VQAAsKeyValuePairDataset)
+            self.assertEqual(kvp_dataset.dataset_info.type, DatasetTypes.KEY_VALUE_PAIR)
+            self.assertIn("name", kvp_dataset.dataset_info.schema)
+            self.assertIn("description", kvp_dataset.dataset_info.schema)
+            self.assertIn("fieldSchema", kvp_dataset.dataset_info.schema)
+
+            self.assertEqual(kvp_dataset.dataset_info.schema["fieldSchema"],
+                             {'answer': {'type': 'string', 'description': 'Answer to the question.'},
+                             'rationale': {'type': 'string', 'description': 'Rationale for the answer.'}})
+
+            _, target, _ = kvp_dataset[0]
+            self.assertIsInstance(target, KeyValuePairLabelManifest)
+            self.assertEqual(target.label_data,
+                             {'fields': {'answer': {'value': 'answer 1'}, 'rationale': {'value': ''}}, 'text': {'question': 'question 1'}})
+
+            self.assertEqual(len(kvp_dataset), 3)
+            self.assertEqual(len(kvp_dataset.dataset_manifest.images), 2)
+
+            # Last image has 2 questions associated with it
+            self.assertEqual(kvp_dataset[-2][0][0].size, kvp_dataset[-1][0][0].size)
```

Comment:
It seems all images are the same with size (100, 100). Could we change the testcase and make the assert effective to check that the last two samples have the same image?

Reply:
Actually, I changed the logic to create image of different sizes for VQA to test exactly this. 
The second image is 200px and so on.